src/many-abci/src/module.rs

```diff
@@ -102,6 +119,79 @@ fn _tm_query_from_many_filter(
     Ok(query)
 }
 
+fn transform_list_args(
+    blockchain::ListArgs {
+        count,
+        order,
+        filter,
+    }: blockchain::ListArgs,
+) -> Result<(u64, Order, Query), ManyError> {
+    filter
+        .map_or(
+            Ok(DEFAULT_BLOCK_LIST_QUERY.clone()),
+            _tm_query_from_many_filter,
+        )
+        .map(|filter| {
+            (
+                count.map_or(MAXIMUM_BLOCK_COUNT, |c| {
+                    std::cmp::min(c, MAXIMUM_BLOCK_COUNT)
+                }),
+                order.map_or(tendermint_rpc::Order::Ascending, _tm_order_from_many_order),
+                filter,
+            )
+        })
+}
+
+fn create_pagination(count: u64) -> (u64, u64) {
```

Comment:
`count` should be a `u8` and you should return a `u8` as `count`

Reply:
@fmorency `count` doesn't need to be downcasted to a `u8` until it is placed as an argument to `tx_search`.  I purposefully have deferred the downcast until the last moment, which simplifies other calculations using it beforehand.